test/http_server_test2/http_server_test2.cpp

```diff
@@ -0,0 +1,363 @@
+                                                                               "Request parameter 'size' must be > 0"),
+                                   false);
+                }
+                else
+                {
+                    response.reply(std::make_unique<SendBlob>(size), false);
+                }
+            }
+        };
+
+        auto upload = [web_root, uploads](
+                IServerResponse& response,
+                IConnectionTimeoutModifier& timeout_modifier,
+                const std::string& url,
+                bool first_part,
+                bool last_part,
+                const std::unordered_map<std::string, std::string>& headers,
+                const std::unordered_map<std::string, std::string>& request_parameters,
+                const std::vector<uint8_t>& content,
+                MIMEParser& mime) {
+            (void) timeout_modifier;
+            (void) headers;
+            (void) request_parameters;
+            (void) url;
+
+            if (first_part)
+            {
+                // Prepare mime parser to receive data
+                mime.detect_mode(headers.at(CONTENT_TYPE), std::stoul(headers.at(CONTENT_LENGTH)));
+            }
+
+            auto form_data = [uploads, &last_part, &response](const std::string& name,
+                                                        const std::string& actual_file_name,
+                                                        const MIMEParser::BoundaryIterator& begin,
+                                                        const MIMEParser::BoundaryIterator& end,
+                                                        const bool file_start,
+                                                        const bool file_close) {
+                // Store the file in web_root/uploads
+                static Path path{};
+                static std::ofstream to_save;
+
+                if(file_start)
+                {
+                    path = uploads / actual_file_name;
+                    Log::info("form_data", "name: {}; File name: {}", name, path.str());
+                    create_directory(path.parent());
+                    // Log::info("form_data", "directory created");
+                    if (FileInfo{ path }.exists())
+                    {
+                        // Log::info("form_data", "File exists");
+                        remove(path);
+                    }
+                    // Log::info("form_data", "before open");
+                    to_save.open(path.str(), std::ios::out | std::ios::app | std::ios::binary) ;
+                    // Log::info("form_data", "File opened");
+                }
+
+
+                auto len = std::distance(begin, end);
+                Log::info("form_data", "chunk to write: {},", static_cast<int>(len));
+                to_save.write(static_cast<const uint8_t*>(&*begin), static_cast<int>(len));
```

Comment:
This is pretty ugly. The BoundaryIterator can provide a `const uint8_t data()` method for cleaner code.

Alternatively, a helper class that takes the two B-iterators and provides the above method + `size_type len() const`

Reply:
to be honest, I don't get it. If I write: 
`to_save.write((char*)&*begin, static_cast<int>(len));`
the code compiles without errors/warning on my machine; and the code also works. Using data() or reinterpret_cast cause various error messages.
Do you have a more specific hint on how to solve it?

Reply:
I meant `const uint8_t* data()`, but I suppose you figured that out already.

I'm guessing the problem is different compiler versions or settings. I don't have things setup so I can unfortunately not dig deeper into it at this time. The docker image is from Sept 26, 2019 so whatever xtensa-gcc8 was awailable then is what is being used now. IIRC, it was the second version.. "r2" something? Perhaps it is time to update the docker image.